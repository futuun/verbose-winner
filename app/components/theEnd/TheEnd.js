import React, { PropTypes } from 'react'
import styles from './styles.sass'

function TheEnd(props) {
  if (props.loading) {
    return (
      <div className={styles.theEnd}>
        <span className={styles.title}>LOADING</span>
      </div>
    )
  }

  return (
    <div className={styles.theEnd}>
      <span className={styles.title}>GAME OVER</span>
      <button
        className={styles.btn}
        onClick={props.newWord}
        autoFocus
      >
        NEW WORD
      </button>
    </div>
  )
}

TheEnd.defaultProps = {
  newWord: () => false,
  loading: true,
}

TheEnd.propTypes = {
  newWord: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
}

export default TheEnd
