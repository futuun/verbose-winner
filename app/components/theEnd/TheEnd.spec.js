import React from 'react'
import renderer from 'react-test-renderer'
import TheEnd from './TheEnd'

test('TheEnd: loading', () => {
  const component = renderer.create(
    <TheEnd loading />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('TheEnd: game over', () => {
  const component = renderer.create(
    <TheEnd loading={false} />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
