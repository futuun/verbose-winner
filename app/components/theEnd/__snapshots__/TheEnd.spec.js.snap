exports[`test TheEnd: game over 1`] = `
<div
  className="theEnd">
  <span
    className="title">
    GAME OVER
  </span>
  <button
    autoFocus={true}
    className="btn"
    onClick={[Function]}>
    NEW WORD
  </button>
</div>
`;

exports[`test TheEnd: loading 1`] = `
<div
  className="theEnd">
  <span
    className="title">
    LOADING
  </span>
</div>
`;
